csharp/src/Drivers/Databricks/DatabricksOperationStatusPoller.cs

```
@@ -69,13 +72,18 @@ private async Task PollOperationStatus(CancellationToken cancellationToken)
                     var operationHandle = _statement.OperationHandle;
                     if (operationHandle == null) break;
 
```

Use a timeout token here, instead of cancelling when canceltoken is triggered; if an interrupt is triggered prematurely, the TCLI client may still have unconsumed results in the Input buffer

---

we could want to swallow timeout exceptions here though, keep polling unless explicitly canceled by reader class

---

are you able to repro this? should we do this to all the thrift rpc calls in the driver?

---

Yea, during my testing, if stop() canceled the cancellationToken, it would result in some strange behavior, GetOperationStatus issues causing CloseOperation to fail.

I think we are safe in HiveServer2Statement, but we might need to adjust CancellationToken in DatabricksReader, CloudFetchResultFetcher, and DatabricksCompositeReader, we are passing in CancellationToken that user can trigger mid-call

---

Actually, I think this depends a bit on how CancellationToken could be used by PBI, too
@CurtHagenlocher will mashup ever trigger cancellationTokens passed into IArrowStreamReader.ReadNextBatchAsync? Do we need to ensure that the connection still remains usable for subsequent statements?

---

At least for now, I think we can operate this way:
1. If the user cancels the token passed in to ReadNextBatchAsync, we should try not to break the client (we could change on this?)
2. Dispose() should not break the client either